pyop2/sequential.py

```diff
@@ -890,14 +979,70 @@ def extrusion_loop():
     _map_bcs_m = ""
     _map_bcs_p = ""
     _layer_arg = ""
+    _layer_decls = ""
+    _iterset_masks = ""
+    _entity_offset = ""
+    _get_mask_indices = ""
     if itspace._extruded:
-        _layer_arg = ", int start_layer, int end_layer, int top_layer"
+        _layer_arg = ", %s *layers" % as_cstr(IntType)
+        if itspace.iterset.constant_layers:
+            idx0 = "0"
+            idx1 = "1"
+        else:
+            if isinstance(itspace.iterset, Subset):
+                # Subset doesn't hold full layer array
+                idx0 = "2*n"
+                idx1 = "2*n+1"
```

Comment:
I filter the layer array etc in the subset constructor because that seems more natural than only reindexing in the generated code.